dag_in_context/src/optimizations/swap_if.egg

```diff
@@ -0,0 +1,27 @@
+(ruleset swap-if)
+
+;; swaps the order of the then and else branches
+;; in an if using Not
+
+(rule
+  ((= lhs (If pred inputs then else)))
+  (
+    (union lhs (If (Uop (Not) pred) inputs else then))
+  )
+  :ruleset swap-if)
+
+
+;; for if statements with two outputs, swaps the order
+;; of the outputs
+(rule
+  ((= lhs (If pred inputs then else))
+   (= (tuple-length then) 2)
+   (= (tuple-length else) 2))
+  (
+    (union
```

Comment:
Would it be more useful to union `lhs` with `(Concat (Single (Get new-lhs 1)) (Single (Get new-lhs 0)))` where `new-lhs` is the if expression with output swapped? After all, the expression being unioned here, `(Concat (Single (Get lhs 1)) (Single (Get lhs 0)))`, isn't part of the final program.

Reply:
I believe that our tuple helpers, which create gets for all of the tuples, do discover that equivalence